lib/src/simple_token.dart

```diff
@@ -0,0 +1,137 @@
+
+library angular_ast.src.simple_token;
+
+import 'package:quiver/core.dart';
+
+part 'token/simple_type.dart';
+
+class NgSimpleToken {
+  factory NgSimpleToken.bang(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.bang, offset);
+  }
+
+  //Probably don't need
+  factory NgSimpleToken.closeBrace(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeBrace, offset);
+  }
+
+  factory NgSimpleToken.closeBracket(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeBracket, offset);
+  }
+
+  factory NgSimpleToken.closeParen(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeParen, offset);
+  }
+
+  factory NgSimpleToken.commentBegin(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.commentBegin, offset);
+  }
+
+  factory NgSimpleToken.commentEnd(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.commentEnd, offset);
+  }
+
+  factory NgSimpleToken.dash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.dash, offset);
+  }
+
+  factory NgSimpleToken.doubleQuote(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.doubleQuote, offset);
+  }
+
+  factory NgSimpleToken.doubleQuotedText(int offset, String lexeme) {
+    return new NgSimpleToken(
+        new NgSimpleTokenType.doubleQuotedText(lexeme), offset);
+  }
+
+  factory NgSimpleToken.tagStart(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.tagStart, offset);
+  }
+
+  factory NgSimpleToken.tagEnd(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.tagEnd, offset);
+  }
+
+  factory NgSimpleToken.EOF(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.EOF, offset);
+  }
+
+  factory NgSimpleToken.equalSign(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.equalSign, offset);
+  }
+
+  factory NgSimpleToken.forwardSlash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.forwardSlash, offset);
+  }
+
+  factory NgSimpleToken.hash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.hash, offset);
+  }
+
+  //Probably don't need
+  factory NgSimpleToken.openBrace(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openBrace, offset);
+  }
+
+  factory NgSimpleToken.openBracket(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openBracket, offset);
+  }
+
+  factory NgSimpleToken.openParen(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openParen, offset);
+  }
+
+  factory NgSimpleToken.singleQuote(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.singleQuote, offset);
+  }
+
+  factory NgSimpleToken.singleQuotedText(int offset, String lexeme) {
+    return new NgSimpleToken(
+        new NgSimpleTokenType.singleQuotedText(lexeme), offset);
+  }
+
+  factory NgSimpleToken.star(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.star, offset);
+  }
+
+  factory NgSimpleToken.text(int offset, String lexeme) {
```

Comment:
I don't think the tokenizer should be agnostic. A text block will not occur inside a property, and neither will a comment. Moreover, they have different characteristics (a comment block has open & close delimiters, a nodename doesn't have spaces, an attribute name doesn't have dashes...). Its customary to split different types of identifiers...even "ng-content", "select", "on", "bind" etc. could be treated as its own angular "keyword"s so that the parser doesn't accidentally mix them with other non-special identifiers.

The main point here is to let the typesystem catch mistakes in the parser. "token is X" should rule out all garbage possibilities, such as a text node erroneously inside a tag, or a "bind" keyword inside a "[...]" binding. etc etc